basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
             return constrained_points;
         }
 
+        std::vector<lanelet::BasicPoint2d> create_lanechange_route(int ending_lane_id, double starting_downtrack, double ending_downtrack,
+                                                                   const carma_wm::WorldModelConstPtr &wm,const cav_msgs::VehicleState &state)
+        {
+            std::vector<lanelet::BasicPoint2d> centerline_points = {};
+
+            //Get starting lanelet and ending lanelets
+            auto lanelets = wm->getLaneletsBetween(starting_downtrack, ending_downtrack,true);
+            lanelet::ConstLanelet starting_lanelet =lanelets.front();
+            lanelet::ConstLanelet ending_lanelet = wm->getMap()->laneletLayer.get(ending_lane_id);
+            
+            lanelet::ConstLanelet current_lanelet = starting_lanelet;
+            lanelet::ConstLanelets starting_lane;
+            starting_lane.push_back(starting_lanelet);
+
+            std::vector<lanelet::BasicPoint2d> reference_centerline;
+            bool shared_boundary_found = false;
+            bool is_lanechange_left = false;
+            
+            lanelet::BasicLineString2d current_lanelet_centerline = starting_lanelet.centerline2d().basicLineString();
+            reference_centerline.insert(reference_centerline.end(), current_lanelet_centerline.begin(), current_lanelet_centerline.end());
+
+            while(!shared_boundary_found){
+
+                if(current_lanelet.leftBound() == ending_lanelet.rightBound()){   
+                    is_lanechange_left = true;
+                    shared_boundary_found = true;
+                }
+
+                else if(current_lanelet.rightBound() == ending_lanelet.leftBound()){
+                    shared_boundary_found = true;
+                }
+
+                else{
+                    //In case there are no following lanelets on route, lanechange should be completing before reaching it
+                    if(wm->getMapRoutingGraph()->following(current_lanelet, false).empty())
```

Comment:
I think a bit more description on this check would be helpful. As I understand it we could have a lanechange that touches the final lanelet on the route, but presumable the above two checks would trigger on this iteration and so we wouldn't hit this?